New package: py-dask-xgboost

a-saitoh-fj opened the pull request:
New package: py-dask-xgboost

a-saitoh-fj committed:
New package: py-dask-xgboost

adamjstewart commented on var/spack/repos/builtin/packages/py-dask-xgboost/package.py:
```diff
@@ -0,0 +1,26 @@
+
+    variant('sparse', default=False, description='Extras require')
+
+    depends_on('python', type=('build', 'run'))
```
You can remove this line, it's implied by the base class

adamjstewart commented on var/spack/repos/builtin/packages/py-dask-xgboost/package.py:
```diff
@@ -0,0 +1,26 @@
+
+    version('0.1.11', sha256='3fbe1bf4344dc74edfbe9f928c7e3e6acc26dc57cefd8da8ae56a15469c6941c')
+
+    variant('sparse', default=False, description='Extras require')
```
I think a better description would be something like "add sparse support" or something like that

adamjstewart commented on var/spack/repos/builtin/packages/py-dask-xgboost/package.py:
```diff
@@ -0,0 +1,26 @@
+
+    depends_on('python', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('xgboost@:0.90', type=('build', 'run'))
```
```suggestion
    depends_on('py-xgboost@:0.90', type=('build', 'run'))
```

adamjstewart commented:
As mentioned in #21661, according to https://github.com/dask/dask-xgboost/issues/80, `py-dask-xgboost` is actually deprecated, and the functionality has been migrated to `py-xgboost+dask`. Is this package still necessary, or can it be removed?

a-saitoh-fj committed:
py-dask-xgboost: Fixed what was pointed out

a-saitoh-fj commented:
py-dask-xgboost: Fixed what was pointed out

- Deleted the description of "depends_on('python', type=('build', 'run'))"
- Change description
- Fix dependency from xgboost to py-xgboost

Submit this recipe as it may be used in py-dask-ml.
I think it may change in the discussion of #21661, but I will submit it first.
